combat-trainer.lic

```diff
@@ -4830,13 +4833,53 @@ class GameState
     return unless can_engage?
     return if stomp
     return if pounce
+    return if rush
 
     case DRC.bput("engage", /^You (are already|begin to) (stealthily )?(advanc(e|ing)|at melee)/, /^There is nothing else/, /is already quite dead/, /^What do you want to advance towards/)
     when /You are already advancing/, /You begin to advance/, /You begin to stealthily advance/
       pause 2
     end
   end
 
+  def rush
+    return if retreating?
+    return false if DRC.left_hand
+    return false unless @rush_shield
+    return false unless npcs.any?
+    return false unless charged_maneuver_off_cooldown?(@charged_maneuvers['Shield Usage'])
+
+    wield_specific_weapon(@rush_shield, nil, false)
+    maneuver_success = case DRC.bput("maneuver rush",
+                                     /^You angle your .* towards .* and charge forwards/,
+                                     /^You are already engaged/,
+                                     /is already engaged to you/,
+                                     /^You are unable to focus on performing a maneuver while aiming at an enemy/,
+                                     /^There is nothing else to face/,
+                                     /^You must rest a bit longer before attempting/)
+                       when /You must rest a bit longer before attempting/ # ability not actually off cooldown
+                         @cooldown_timers['rush'] += 15
+                         false
+                       when /There is nothing else to face/ # nothing to rush
+                         false
+                       when /You are unable to focus on performing a maneuver while aiming at an enemy/
+                         @cooldown_timers['rush'] += 15 # If we happen to try to engage while aiming, don't retry immediately
+                         false
+                       when /You are already engaged/, /is already engaged to you/ # already at melee
+                         true
+                       when /^You angle your .* towards .* and charge forwards/
+                         if Flags['ct-maneuver-cooldown-reduced'] && DRStats.paladin?
+                           @cooldown_timers['rush'] = Time.now + 45 # Paladins can get a shortened timer (similar to Barbarians for all other maneuvers)
+                           Flags.reset('ct-maneuver-cooldown-reduced')
+                         else
+                           @cooldown_timers['rush'] = Time.now + 90
+                         end
+                         true
+                       end
+
+    DRCI.wear_item?(@rush_shield)
```

Comment:
If `@rush_shield` matches an entry in the character's gear config, I recommend using `@equipment_manager.stow_weapon(@rush_shield)`

Despite the method name, it handles putting away or wearing any gear item. This gives greater flexibility so that characters may opt to pull out a large shield from a container, rush, put it away, then continue combat with their existing arm-worn shield.

If a character wants to use their arm-worn shield, then they will put that shield's name in `@rush_shield` and the script will hold/wear the shield as expected.